src/skin/legacyskinparser.cpp

```diff
@@ -2007,11 +2019,11 @@ void LegacySkinParser::setupConnections(const QDomNode& node, WBaseWidget* pWidg
                                                         pTransformer, property);
             pWidget->addPropertyConnection(pConnection);
 
-            // If we created this control, bind it to the
-            // ControlWidgetConnection so that it is deleted when the connection
-            // is deleted.
+            // If we created this control, add it to the set of skin-created
+            // controls, so that it can be deleted when the MixxxMainWindow is
+            // destroyed.
             if (created) {
-                control->setParent(pConnection);
+                m_pSkinCreatedControls->insert(control);
```

Comment:
Is it possible to avoid all the redundant insert operations and comments by moving the code directly into controlFromConfigNode()?

Reply:
Otherwise we should at least encapsulate it in a separate method.

Reply:
I didn't notice that there is a separate constructor with nullptr. If we could isolate all the bookkeeping in a single method we might think about using VERIFY_OR_DEBUG_ASSERT before the insertion again and abort it with logging a warning.

When inserting the control into the set we should also assert that this does not replace an already existing entry, which would be unexpected.

Reply:
28527e8d9ddec8cbae34a8d5a5d51a7aa0ee4cd1

Reply:
3e0817e86e81fd02a3c4846905c784b278c01fb5